Userland/Libraries/LibGfx/AntiAliasingPainter.cpp

```diff
@@ -27,8 +27,56 @@ void AntiAliasingPainter::draw_anti_aliased_line(FloatPoint const& actual_from,
 
     auto corrected_thickness = thickness > 1 ? thickness - 1 : thickness;
     auto size = IntSize(corrected_thickness, corrected_thickness);
+    auto mapped_from = m_transform.map(actual_from);
+    auto mapped_to = m_transform.map(actual_to);
+
+    auto draw_direction = mapped_from - mapped_to;
+    auto is_straight_line = draw_direction.x() == 0 || draw_direction.y() == 0;
+
+    FloatQuad rotated_rectangle_reference_coords = FloatQuad({ 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 });
+
+    float drawing_edge_offset = 0;
+
+    if (!is_straight_line) {
```

Comment:
Swap these two and then skip the 'else' branch and move it out (as the if part returns).